app/admin/quests/dashboard/[questId]/page.tsx

```diff
@@ -0,0 +1,740 @@
+"use client";
+
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import styles from "@styles/admin.module.css";
+import { useRouter } from "next/navigation";
+import { AdminService } from "@services/authService";
+import { QuestDefault } from "@constants/common";
+import { nft_uri, QuizQuestionDefaultInput, formSteps } from "@constants/admin";
+import {
+  NFTUri,
+  UpdateBoost,
+  UpdateQuest,
+  UserTask,
+} from "../../../../../types/backTypes";
+import AdminQuestDetails from "@components/admin/questDetails";
+import { useNotification } from "@context/NotificationProvider";
+import { getExpireTimeFromJwt } from "@utils/jwt";
+import Typography from "@components/UI/typography/typography";
+import QuestDetailsForm from "@components/admin/formSteps/QuestDetailsForm";
+import RewardDetailsForm from "@components/admin/formSteps/RewardDetailsForm";
+import TaskDetailsForm from "@components/admin/formSteps/TaskDetailsForm";
+import { TEXT_TYPE } from "@constants/typography";
+import FormContainer from "@components/admin/FormContainer";
+
+type QuestIdProps = {
+  params: {
+    questId: string;
+  };
+};
+
+type WithNewField<T, K extends string, V> = T & { [P in K]: V };
```

Comment:
Looks like Japanese to me. Is there a way to make this clearer or add a comment?

Reply:
done